### PR TITLE
[NFC][EarlyIfConverter] Replace boolean Predicate for a class

### PR DESCRIPTION
**Pull request by jmmartinez**
Currently SSAIfConv is used in 2 scenarios. Generalize them to support more scenarios.

### CODE REVIEW

File: llvm/lib/CodeGen/EarlyIfConversion.cpp [hunk 1]
```diff
@@ -122,6 +122,17 @@ class SSAIfConv {
   /// The branch condition determined by analyzeBranch.
   SmallVector<MachineOperand, 4> Cond;
 
+  struct PredicationStrategyBase {
```
Comment (tschuett) [inline llvm/lib/CodeGen/EarlyIfConversion.cpp:125]: `struct` with inheritance, virtual methods, and `=0`  is really confusing.

Comment (jmmartinez) [reply llvm/lib/CodeGen/EarlyIfConversion.cpp]: I agree. Updated to `class` for `PredicationStrategyBase` and its derived classes.

**Comment by jmmartinez** [comment]
If this PR is ok for you I'd like to merge it.

**Comment by arsenm** [comment]
Wasn't reviewed?

**Comment by jmmartinez** [comment]
> Wasn't reviewed?

Sorry, I've assumed it was. I've addressed the remarks and after some time without new feedback I merged it.

**Comment by arsenm** [comment]
Are you still planning on using this? I thought you abandoned the speculative execution changes. If so should this be reverted?

**Comment by jmmartinez** [comment]
I'll revert them. I don't think we need them anymore.

**Pull request merged**